log_cache.go

```diff
@@ -0,0 +1,98 @@
+package raft
+
+import (
+	"sync"
+)
+
+// logCache wraps a logstore with a ring buffer providing fast access to the
+// last n raft log entries.
+type logCache struct {
```

Comment:
I think we should expose this and allow a caller to wrap a LogStore of their choice. It might be the case that the LogStore is doing this internally already and there is no need to double cache.